File: Data Structures & Algorithms/eating-bananas/submission-0.py
import math
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        a, r = 1, max(piles)
        res = r
        while a <= r:
            hour = 0
            n = (r + a) // 2
            for e in piles:
                hour += math.ceil(e / n)
            if hour > h:
                a = n + 1
            elif hour <= h:
                res = min(res, n)
                r = n - 1
        return res

File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int i = 0;
        int j = heights.size() - 1;
        int max = 0;
        int area;
        while (i < j){
            area = min(heights[i], heights[j]) * (j - i);
            if (area > max)
                max = area;
            if (heights[i] > heights[j])
                j -= 1;
            else
                i += 1;
        }
        return max;
    }
};
